@using Microsoft.AspNetCore.Components.Authorization
@inject NavigationManager NavigationManager

<CascadingAuthenticationState>
    <CustomErrorBoundary @ref="errorBoundary">
        <ChildContent>
            <Router AppAssembly="@typeof(Program).Assembly">
                <Found Context="routeData">
                    <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)">
                        <NotAuthorized>
                            @if (!NavigationManager.Uri.Contains("/login"))
                            {
                                NavigationManager.NavigateTo("/login", forceLoad: true);
                            }
                        </NotAuthorized>
                    </AuthorizeRouteView>
                </Found>
                <NotFound>
                    <LayoutView Layout="@typeof(MainLayout)">
                        <NotFound />
                    </LayoutView>
                </NotFound>
            </Router>
        </ChildContent>
        <ErrorContent>
            <p class="m-4 text-red-600">Произошла непредвиденная ошибка.</p>
        </ErrorContent>
    </CustomErrorBoundary>
</CascadingAuthenticationState>

<ErrorDialog />

@code {
    CustomErrorBoundary? errorBoundary;
}
